test/test_functional.py

```diff
@@ -0,0 +1,182 @@
+import math
+import os
+
+import torch
+import torchaudio
+import unittest
+import test.common_utils
+
+
+class TestFunctional(unittest.TestCase):
+    data_sizes = (2, 20)
+    number_of_trials = 100
+
+    def _compare_estimate(self, sound, estimate, atol=1e-6, rtol=1e-8):
+        # trim sound for case when constructed signal is shorter than original
+        sound = sound[..., :estimate.size(-1)]
+
+        self.assertTrue(sound.shape == estimate.shape, (sound.shape, estimate.shape))
+        self.assertTrue(torch.allclose(sound, estimate, atol=atol, rtol=rtol))
+
+    def _test_istft_is_inverse_of_stft(self, kwargs):
+        # generates a random sound signal for each tril and then does the stft/istft
+        # operation to check whether we can reconstruct signal
+        for i in range(self.number_of_trials):
+            sound = torch.rand(self.data_sizes)
+
+            stft = torch.stft(sound, **kwargs)
+            estimate = torchaudio.functional.istft(stft, length=sound.size(1), **kwargs)
+
+            self._compare_estimate(sound, estimate)
+
+    def test_istft_is_inverse_of_stft1(self):
+        # hann_window, centered, normalized, onesided
+        kwargs1 = {
+            'n_fft': 12,
+            'hop_length': 4,
+            'win_length': 12,
+            'window': torch.hann_window(12),
+            'center': True,
+            'pad_mode': 'reflect',
+            'normalized': True,
+            'onesided': True,
+        }
+
+        self._test_istft_is_inverse_of_stft(kwargs1)
+
+    def test_istft_is_inverse_of_stft2(self):
+        # hann_window, centered, not normalized, not onesided
```

Comment:
Did you also add cases for when the windows don't overlap?

Reply:
yes test_istft_requires_overlap_windows